src/toast/qarray.py

```diff
@@ -412,6 +414,122 @@ def from_vectors(v1, v2):
         return out.array().reshape((-1, 4))
 
 
+def from_iso(theta, phi, psi):
+    """Create quaternions from ISO theta, phi, psi spherical coordinates.
+
+    The input angles describe the ZYZ rotations used to build the quaternion.
+
+    Args:
+        theta (array):  Array or scalar theta values in radians
+        phi (array):  Array or scalar phi values in radians
+        psi (array):  Array or scalar psi values in radians
+
+    Returns:
+        (array):  The quaternions.
+
+    """
+    intheta = ensure_buffer_f64(theta)
+    inphi = ensure_buffer_f64(phi)
+    inpsi = ensure_buffer_f64(psi)
+    lt = len(intheta)
+    out = AlignedF64(4 * lt)
+    qa_from_iso(intheta, inphi, inpsi, out)
+    if len(out) == 4:
+        if (
+            (object_ndim(theta) == 1)
+            or (object_ndim(phi) == 1)
+            or (object_ndim(psi) == 1)
+        ):
+            return out.array().reshape((1, 4))
+        else:
+            return out.array()
+    else:
+        return out.array().reshape((-1, 4))
+
+
+def from_lonlat(lon, lat, psi):
+    """Create quaternions from longitude, latitude, and psi spherical coordinates.
+
+    Args:
+        lon (array):  Array or scalar lon values in radians
+        lat (array):  Array or scalar lat values in radians
+        psi (array):  Array or scalar psi values in radians
+
+    Returns:
+        (array):  The quaternions.
+
+    """
+    inlon = ensure_buffer_f64(lon)
+    inlat = ensure_buffer_f64(lat)
+    inpsi = ensure_buffer_f64(psi)
+    lt = len(inlon)
+    out = AlignedF64(4 * lt)
+    theta = AlignedF64(lt)
+    theta[:] = 0.5 * np.pi - np.array(inlat)
+    qa_from_iso(theta, inlon, inpsi, out)
+    if len(out) == 4:
+        if (
+            (object_ndim(lon) == 1)
+            or (object_ndim(lat) == 1)
+            or (object_ndim(psi) == 1)
+        ):
+            return out.array().reshape((1, 4))
+        else:
+            return out.array()
+    else:
+        return out.array().reshape((-1, 4))
+
+
+def to_iso(q):
```

Comment:
I remember using `to_angles()` a few times. Did you search and replace all instances of the method?

Reply:
I believe so- although I also fixed that in the separately in the sotodlib toast3 branch